modules/oxide/src/less/theme/components/toggle/toggle.less

```
@@ -1,19 +1,26 @@
 .tox {
-  .tox-toggle {
-    // private variables scoped to tox-toggle class
-    --tox-private-slider-background-color: @background-color;
-    --tox-private-slider-border-color: .bg-luma-checker(hsl(from @color-white h s calc(l - 11)), hsla(from @color-white h s l / 15%))[@result]; // to be figured out after the spike with conditionals in CSS
-    --tox-private-slider-handle-background-color: @text-color;
+  .tox-toggle when (@custom-properties-enabled = true) {
+    --tox-private-slider-background-color: var(--tox-private-background-color);
+    --tox-private-slider-border-color: light-dark(
+      hsl(from var(--tox-private-color-white) h s calc(l - 11)), 
+      hsl(from var(--tox-private-color-white) h s l / 15%));
```

As mentioned previously, there were compile issue with newer LESS, specifically on `from...` syntax. Moving this around won't solve it. It doesn’t matter that it’s inside a CSS variable. It doesn’t matter that it’s inside `.bg-luma-checker(...)`. As long as Less tries to treat that as a color and do math on it, it’ll blow up with `invalid type`.

`hsl(...) / hsla(...)` are parsed as Less color functions, but `from @color-white h s l...` is not a valid Less signature.

---

I found a reference that suggests doing this to bypass the Less compiler and emit it as a string. I haven't checked if it works.
```suggestion
      ~"hsl(from var(--tox-private-color-white) h s calc(l - 11))", 
      ~"hsl(from var(--tox-private-color-white) h s l / 15%))";
```

---

yeah, if we need to dump the css here, while keeping less, it'll have to be the bypass, perhaps something like this?

```suggestion
  --tox-private-slider-border-color: ~"light-dark(
    hsl(from var(--tox-private-color-white) h s calc(l - 11)), 
    hsl(from var(--tox-private-color-white) h s l / 15%)
  )";
```

---

Yes, it's from bumping less.

This might not be an issue if we using CSS directly, tho I'm not sure if it's the best combination of syntax to use 🤔

---

I agree, is there scope to do that in this PR?